ChaosMod/Effects/db/Misc/MiscGetTowed.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+Effect by Last0xygen
+*/
+
+#include <stdafx.h>
+
+
+static void OnStart()
+{
+	static const Hash tonyaHash = GET_HASH_KEY("ig_tonya");
+	static const Hash towTruckHash = GET_HASH_KEY("towtruck");
+	static Hash relationshipGroup;
+	ADD_RELATIONSHIP_GROUP("_TOW_TRUCK_TONYA", &relationshipGroup);
+	SET_RELATIONSHIP_BETWEEN_GROUPS(0, relationshipGroup, GET_HASH_KEY("PLAYER"));
+	LoadModel(tonyaHash);
+	LoadModel(towTruckHash);
+
+	Vector3 rearBottomLeft, frontTopRight;
+	GET_MODEL_DIMENSIONS(towTruckHash, &rearBottomLeft, &frontTopRight);
+	float towLength = frontTopRight.y - rearBottomLeft.y;
+
+	Ped player = PLAYER_PED_ID();
+	Vehicle playerVeh = 0;
+	float spawnOffset = towLength / 2;
+	if (IS_PED_IN_ANY_VEHICLE(player, false))
+	{
+		playerVeh = GET_VEHICLE_PED_IS_IN(player, false);
+		Hash vehMod = GET_ENTITY_MODEL(playerVeh);
+		GET_MODEL_DIMENSIONS(vehMod, &rearBottomLeft, &frontTopRight);
+		float playerVehLength = frontTopRight.y - rearBottomLeft.y;
+		spawnOffset = (towLength / 2) + (playerVehLength / 2) + 1;
+	}
+
+
+	Vector3 spawnPoint = GET_OFFSET_FROM_ENTITY_IN_WORLD_COORDS(player, 0, spawnOffset, 0);
```

Comment:
Maybe check if there is a wall right in front of the player, and spawn the truck behind the player? Otherwise things like this may happen 😅
![image](https://user-images.githubusercontent.com/6181929/122999149-444d6a00-d384-11eb-9a3c-12aa34435941.png)